python/tank/context.py

```diff
@@ -1643,8 +1643,6 @@ def _build_entity_dict_from_path(tk, path, required_fields=None, additional_type
     # Grab all project roots
     project_roots = tk.pipeline_configuration.get_data_roots().values()
 
-    # note - paths returned by get_paths are always prefixed with a
-    # project root so there is no risk we end up with an infinite loop here..
```

Comment:
Seems like this was meant to be called only by arguments from get_paths(), but we've added context_from_path which could potentially contain any path.